src/main/java/com/hotels/bdp/circustrain/bigquery/table/service/partitioned/PartitionedTableService.java

```diff
@@ -68,27 +68,16 @@ public Table getTable() {
 
   @Override
   public List<Partition> getPartitions() {
-    return factory.generate(getPartitionKey(filteredTable), result.iterateAll());
+    Field partitionField = getPartitionField(filteredTable.getDefinition().getSchema());
+    return factory.generate(partitionField.getName(), partitionField.getType().name(), result.iterateAll());
   }
 
-  private String getPartitionKey(com.google.cloud.bigquery.Table table) {
-    Schema schema = table.getDefinition().getSchema();
-    if (schema == null) {
-      return "";
+  private Field getPartitionField(Schema schema) {
+    if (filteredTable.getDefinition().getSchema() == null) {
+      throw new IllegalStateException("Could not filter table with the configured filter");
```

Comment:
to get back to your comment (which somehow got lost on me, github is confusing). 
From what I understand you can't be sure if the schema can actually be null. I guess that depends on the api and the code, looks like `getSchema` is annotated with @Nullable, so yes the null check is in order.
I think the exception is ok, just change the message: `"No schema defined for "+fillteredTable.getFriendlyName()+" can't generate partitions"` ??